### PR TITLE
Removes unnecessary calls to ValueProvider.isAccessible

### PR DESCRIPTION
**Pull request by jkff**
Most of this cleanup is in DisplayData. DisplayData.item() already can take a ValueProvider and handles it properly.

R: @lukecwik

### CODE REVIEW

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/display/DisplayData.java [hunk 1]
```diff
@@ -883,7 +883,7 @@ private DisplayData build() {
       }
       Type type = inferType(got);
       if (type == null) {
-        throw new RuntimeException(String.format("Unknown value type: %s", got));
+        return item(key, Type.STRING, got.toString());
```
Comment (lukecwik) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/display/DisplayData.java]: I'm not sure if we want to always assume string type when we fail to infer the type

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java [hunk 2]
```diff
@@ -634,10 +634,6 @@ private static Query translateGqlQuery(String gql, Datastore datastore, String n
     public void validate(PipelineOptions options) {
       checkNotNull(getProjectId(), "projectId");
 
-      if (getProjectId().isAccessible() && getProjectId().get() == null) {
```
Comment (lukecwik) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java]: What if the person passed in a StaticValueProvider containing a null?

**Comment by lukecwik** [comment]
R: @swegner

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/DefaultFilenamePolicy.java [hunk 3]
```diff
@@ -368,26 +368,12 @@ private String paneInfoToString(PaneInfo paneInfo) {
...
-
-    String outputPrefixString = null;
-    outputPrefixString =
-        params.baseFilename.isAccessible()
-            ? params.baseFilename.get().toString()
-            : params.baseFilename.toString();
-    builder.add(DisplayData.item("filenamePattern", filenamePattern).withLabel("Filename Pattern"));
-    builder.add(DisplayData.item("filePrefix", outputPrefixString).withLabel("Output File Prefix"));
-    builder.add(DisplayData.item("fileSuffix", params.suffix).withLabel("Output file Suffix"));
+    builder.add(
```
[the remark this answers is not in the document]
Comment (jkff) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/io/DefaultFilenamePolicy.java]: Done

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSource.java [hunk 4]
```diff
@@ -254,10 +249,6 @@ public void populateDisplayData(DisplayData.Builder builder) {
     // split a FileBasedSource based on a file pattern to FileBasedSources based on full single
     // files. For files that can be efficiently seeked, we further split FileBasedSources based on
     // those files to FileBasedSources based on sub ranges of single files.
-    checkState(
```
[the remark this answers is not in the document]
Comment (jkff) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSource.java]: It doesn't seem more useful than the message one would get from `.get()` anyway

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSource.java [hunk 5]
```diff
@@ -326,10 +317,6 @@ protected boolean isSplittable() throws Exception {
   public final BoundedReader<T> createReader(PipelineOptions options) throws IOException {
     // Validate the current source prior to creating a reader for it.
     this.validate();
-    checkState(
```
[the remark this answers is not in the document]
Comment (jkff) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSource.java]: Ditto

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/display/DisplayData.java (hunk 1 above)
Comment (jkff) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/display/DisplayData.java]: We already assume string type if it's not accessible (we return a `STRING` containing the whole provider's `toString`), so I think it's not too big of a stretch to also do that when we can't infer the type of the stored value.

Main reason I did this change is that I wanted this function to work fine for things like `ValueProvider<ResourceId>` without the user needing to do the `isAccessible` check and doing a `toString` themselves anyway in all cases.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java (hunk 2 above)
Comment (jkff) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java]: I'm not sure we need to be this diligent about nullness checking, but okay, reverted this part.

**Comment by reuvenlax** [comment]
@swegner any comments on this PR?

**Comment by swegner** [comment]
Thanks for the ping. The change LGTM. I like the more concise NestedValueProvider.toString() representation.

**Pull request closed without merging**